Library.java

```diff
@@ -0,0 +1,261 @@
+    }
+
+
+    private int getNextLoanId() {
+        return loanId++;
+    }
+
+
+    public List<member> getMemberList() {
+        Collection<member> membersInMap = members.values();
+        return new ArrayList<member>(membersInMap);
+    }
+
+
+    public List<Book> getBookList() {
+        Collection<Book> booksInMap = catalog.values();
+        return new ArrayList<Book>(booksInMap);
+    }
+
+
+    public List<loan> getCurrentLoansList() {
+        Collection<loan> loansInMap = currentLoans.values();
+        return new ArrayList<loan>(loansInMap);
+    }
+
+
+    public member addMember(String lastName, String firstName, String email, int phoneNo) {
+        int nextMemberId = this.getNextMemberId();
+        member newMember = new member(lastName, firstName, email, phoneNo, nextMemberId);
+        members.put(newMember.GeT_ID(), newMember);
+        return newMember;
+    }
+
+
+    public Book addBook(String a, String t, String c) {
+        int nextBookId = this.getNextBookId();
+        Book newBook = new Book(a, t, c, nextBookId);
+        catalog.put(newBook.getBookId(), newBook);
+        return newBook;
+    }
+
+
+    public member getMemberId(int memberId) {
+        if (members.containsKey(memberId))
+            return members.get(memberId);
+        return null;
+    }
+
+
+    public Book getBookId(int bookId) {
+        if (catalog.containsKey(bookId))
+            return catalog.get(bookId);
+        return null;
+    }
+
+
+    public int getLoanLimit() {
+        return LOAN_LIMIT;
+    }
+
+
+    public boolean canMemberBorrow(member currentMember) {
+        if (currentMember.Number_Of_Current_Loans() == LOAN_LIMIT)
+            return false;
+
+        if (currentMember.Fines_OwEd() >= MAX_FINES_OWED)
+            return false;
+
+        for (loan loanBeingAssessed : currentMember.GeT_LoAnS())
+            if (loanBeingAssessed.OVer_Due())
+                return false;
+
+        return true;
+    }
+
+
+    public int getNumberOfLoansAvailable(member currentMember) {
+        return LOAN_LIMIT - currentMember.Number_Of_Current_Loans();
+    }
+
+
+    public loan issueLoan(Book currentBook, member currentMember) {
+        Date dueDate = Calendar.getInstance().getDueDate(LOAN_PERIOD);
+        int nextLoanId = this.getNextLoanId();
+        loan newLoan = new loan(nextLoanId, currentBook, currentMember, dueDate);
+        currentMember.Take_Out_Loan(newLoan);
+        currentBook.borrowBook();
+        int newLoanId = newLoan.ID();
+        loans.put(newLoanId, newLoan);
+        int currentBookId = currentBook.getBookId();
+        currentLoans.put(currentBookId, newLoan);
+        return newLoan;
+    }
+
+
+    public loan getLoanByBookId(int bookId) {
+        if (currentLoans.containsKey(bookId)) {
+            return currentLoans.get(bookId);
+        }
+        return null;
+    }
+
+
+    public double calculateOverDueFine(loan currentLoan) {
+        if (currentLoan.OVer_Due()) {

```

Comment:
All method names are to start with a lowercase letter and to be in camelBack. Name should also contain verb phrases e.g. isOverDue()

Reply:
Method 'OVer_Due()' belongs to loan.java, not Library.java.